static-site/components/nav/user-or-login-link.tsx

```diff
@@ -0,0 +1,19 @@
+"use client";
```

Comment:
Is this a client component only to be able to use the `UserContext`? Would this be an opportunity to just completely remove the `UserContext` and `UserDataWrapper` component and switch to getting this data server side?

Reply:
@tsibley did this in his previous demo of migrating to Next.js: 

1. https://github.com/nextstrain/nextstrain.org/commit/c98760c sets up the `getCurrentUser` function on the server side 
2. https://github.com/nextstrain/nextstrain.org/commit/164579fe9653bc5d5f9684efc1ed918cd4665b91 uses the server function in the nav bar
3. https://github.com/nextstrain/nextstrain.org/commit/65a62ee06cda7f27ad651971b4c7d7bdef5a1e28 uses the server function for the whoami page

Reply:
Yes, this is a client component so that it can get access to the context. 

I elected to port the context layers over because it felt like a smaller change, like it was more idiomatic Next.js style, and because the scale of changing the contents of the `request` object felt like it was expanding the already large scope here. 

I think if we want to try to get away from using React context for this, that should be a follow-up change, not part of this conversion.

Reply:
That's fair, this has been a long thread to pull on!